src/v/kafka/server/handlers/create_partitions.cc

```diff
@@ -159,6 +159,19 @@ ss::future<response_ptr> create_partitions_handler::handle(
             model::topic_namespace(model::kafka_namespace, tp.name));
       });
 
+    valid_range_end = validate_range(
+      request.data.topics.begin(),
+      valid_range_end,
+      std::back_inserter(resp.data.results),
+      error_code::invalid_request,
+      "Partition count must be greater then current number of partitions",
```

Comment:
seems fine, but don't we already enforce this at the controller?

Reply:
Slightly below this commit we validate similar things such as partition count being greater then 0 and non empty assignments, i felt this belonged here with the other types of error validation